brave/browser-laptop
# Ensure all icons in bookmarks bar are vertically aligned

Ensure all icons in bookmarks bar are vertically aligned
Fix #13789

Also restore different bookmarks bar height when in text-only mode vs with favicons.
Fix #13985

Reduce flicker of bookmark bar / overflow-indicator / last item by having aphrodite not remove the custom class-name on render.

Reduce complexity of bookmarks toolbar state by passing through ToolbarMode 'enum' to component.
Future-proof bookmark mode setting retrieval for state-selector-memoization by passing through appState

## Submitter Checklist:


## Test Plan:
1. Preferences > Bookmarks Bar > choose `Show favicons only`
2. Add 10 - 20 bookmarks to bookmarks bar (or pre-existing profile)
3. Bookmark the New Tab page (as it does not have a favicon)
4. Add a couple of folders to bookmarks bar
5. Extend window width so that there is no overflow indicator
6. Contract window width so that there is an overflow indicator
   - **Alignment of icons for folders, default favicon (New Tab page) and favicons should not differ whether there is the overflow indicator or not.**

7. Preferences > Bookmarks Bar > choose `Text only`
    - **Bookmarks bar should decrease in height by a couple of pixels**



## Reviewer Checklist:


Tests
